internal/compiler/generator/rust.rs

```diff
@@ -1023,12 +1023,22 @@ fn generate_item_tree(
         (None, None)
     };
 
-    let parent_item_index = parent_ctx.and_then(|parent| {
-        parent
-            .repeater_index
-            .map(|idx| parent.ctx.current_sub_component.unwrap().repeated[idx].index_in_tree)
+    let parent_item_expression = parent_ctx.and_then(|parent| {
+        parent.repeater_index.map(|idx| {
+            let sub_component_offset = parent.ctx.current_sub_component.unwrap().repeated[idx].index_in_tree;
+
+            quote!(if let Some((parent_component, parent_index)) = self
+                .parent
+                .clone()
+                .upgrade()
+                .map(|sc| (VRcMapped::origin(&sc), sc.tree_index_of_first_child.get()))
+            {
+                *result = slint::re_exports::ItemRc::new(parent_component, parent_index as usize + #sub_component_offset - 1)
+                    .downgrade();
+            })
+        })
     });
-    let parent_item_index = parent_item_index.iter();
+    let parent_item_expression = parent_item_expression.iter();
```

Comment:
```suggestion
```

This doesn't need to be an iterator  if you don't add more to the expention in the quote! macro.  (quote expends None to nothing)

(Also need to remove the `#(` and `)*`  of course)

Reply:
Good point, fixed!